### ISSUE TITLE
Possible to run multiple experiments on the same machine? 

### ISSUE THREAD

**Issue by villanuevab** [junyanz/pytorch-CycleGAN-and-pix2pix#664]
I have 8 CPUs and 4 GPUs on a Google Cloud VM. I want to be able to run 4 different experiments (allocate 1 GPU per experiment). Right now the CPUs don't seem to be doing much during training. But training is very slow. My `__getitem__` for my custom data has to load multiple .npy files, normalize them and concatenate them. I wonder if this is what's causing training to be so slow. 

Also, I am using `batch_size=1`. There was no increased GPU memory usage when trying to increase batch size to 2 or 4. 

![htop](https://user-images.githubusercontent.com/3021974/58898407-5bdb2c80-86af-11e9-929b-3aa170dd5643.png)

```
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 410.72       Driver Version: 410.72       CUDA Version: 10.0     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  Tesla V100-SXM2...  Off  | 00000000:00:04.0 Off |                    0 |
| N/A   42C    P0    68W / 300W |   2011MiB / 16130MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   1  Tesla V100-SXM2...  Off  | 00000000:00:05.0 Off |                    0 |
| N/A   39C    P0    47W / 300W |   2046MiB / 16130MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   2  Tesla V100-SXM2...  Off  | 00000000:00:06.0 Off |                    0 |
| N/A   41C    P0    59W / 300W |   2008MiB / 16130MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   3  Tesla V100-SXM2...  Off  | 00000000:00:07.0 Off |                    0 |
| N/A   42C    P0    59W / 300W |   2028MiB / 16130MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0     11593      C   python                                      1997MiB |
|    1     11525      C   python                                      2035MiB |
|    2     11519      C   python                                      1997MiB |
|    3     11451      C   python                                      2017MiB |
+-----------------------------------------------------------------------------+
```

Do you have any config tips for how to run multiple experiments concurrently, given this setup? 

Playing around with num_threads doesn't seem to actually allocate the specified number of processors

**Comment by junyanz** [comment]
`num_threads` should help ideally. Maybe you want to time your data loading code. Also, check out your file system IO speed.
